frontend/public/components/search.tsx

```diff
@@ -0,0 +1,253 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { Button, Chip, ChipGroup, ChipGroupToolbarItem, Expandable } from '@patternfly/react-core';
+import { CloseIcon } from '@patternfly/react-icons';
+
+import { AsyncComponent } from './utils/async';
+import { connectToModel } from '../kinds';
+import { DefaultPage } from './default-resource';
+import { requirementFromString } from '../module/k8s/selector-requirement';
+import { ResourceListDropdown } from './resource-dropdown';
+import { resourceListPages } from './resource-pages';
+import { withStartGuide } from './start-guide';
+import { split, selectorFromString } from '../module/k8s/selector';
+import { referenceForModel, kindForReference } from '../module/k8s';
+import { LoadingBox, PageHeading, ResourceIcon } from './utils';
+import { SearchFilterDropdown, searchFilterValues } from './search-filter-dropdown';
+import { setQueryArgument } from './utils/router';
+
+const ResourceList = connectToModel(({ kindObj, mock, namespace, selector, nameFilter }) => {
+  if (!kindObj) {
+    return <LoadingBox />;
+  }
+
+  const componentLoader = resourceListPages.get(referenceForModel(kindObj), () =>
+    Promise.resolve(DefaultPage),
+  );
+  const ns = kindObj.namespaced ? namespace : undefined;
+
+  return (
+    <AsyncComponent
+      loader={componentLoader}
+      namespace={ns}
+      selector={selector}
+      nameFilter={nameFilter}
+      kind={kindObj.crd ? referenceForModel(kindObj) : kindObj.kind}
+      showTitle={false}
+      hideTextFilter
+      autoFocus={false}
+      mock={mock}
+    />
+  );
+});
+
+const SearchPage_: React.FC<SearchProps> = (props) => {
+  const [selectedItems, setSelectedItems] = React.useState(new Set<string>([]));
+  const [collaspedState, setCollaspedState] = React.useState(new Set<string>([]));
+  const [selector, setSelector] = React.useState(selectorFromString(''));
+  const [nameFilter, setNameFilter] = React.useState([]);
+  const [labelFilter, setLabelFilter] = React.useState([]);
+
+  const { namespace, noProjectsAvailable } = props;
+
+  // Set state variables from the URL
+  React.useEffect(() => {
+    let kind: string, q: string, name: string;
```

Comment:
You may be able to simplify/`const` some of this with destructuring and eliminate some conditional logic:

```typescript 
const {kind, q, name} = window.location.search 
    ? queryFromSearch() 
    : { kind: '', q: '', name: ''};
```

Reply:
I like what you are doing here, but I would have to make the `queryFromSearch` function and that would be just as many lines as before and so we would not really save anything.